web/src/settings_profile_fields.ts

```diff
@@ -752,10 +843,26 @@ function set_up_select_field(): void {
         },
     );
 
-    $profile_field_choices.on("input", ".choice-row input", add_choice_row);
+    $profile_field_choices.on(
+        "input",
+        ".choice-row input",
+        function (this: HTMLElement, event: JQuery.TriggeredEvent) {
+            add_choice_row.call(this, event);
+            maybe_toggle_alphabetize_button($profile_field_choices);
+        },
+    );
     $profile_field_choices.on("click", "button.delete-choice", function (this: HTMLElement) {
         delete_choice_row(this);
+        maybe_toggle_alphabetize_button($profile_field_choices);
```

Comment:
If the button is already disabled, indicating that the fields are sorted, then we should not call this function when deleting a choice, as deleting a choice from sorted fields will still keep them sorted. Right?

Reply:
Yep, you are right! Thanks for finding this optimization!

I have added a parameter and the actual check inside `maybe_toggle_alphabetize_button` function. This is because we already have a reference to the button in `maybe_toggle_alphabetize_button`.

```diff
diff --git a/web/src/settings_profile_fields.ts b/web/src/settings_profile_fields.ts
index 12cbad1746..9f2d2e62fa 100644
--- a/web/src/settings_profile_fields.ts
+++ b/web/src/settings_profile_fields.ts
@@ -440,7 +440,19 @@ function alphabetize_profile_field_choices(
     $("#edit-custom-profile-field-form-modal .dialog_submit_button").prop("disabled", false);
 }
 
-function maybe_toggle_alphabetize_button($sortable_element: JQuery): void {
+function maybe_toggle_alphabetize_button(
+    $sortable_element: JQuery,
+    action_unsorts_field = true,
+): void {
+    // This is safe because only one profile field modal is open at a time ever.
+    const $alphabetize_choices_button = $(
+        ".profile-field-choices-wrapper > button.alphabetize-choices-button",
+    );
+
+    if (!action_unsorts_field && $alphabetize_choices_button.prop("disabled")) {
+        // The fields are already sorted.
+        return;
+    }
     assert($sortable_element[0] !== undefined);
     const sortable_instance = SortableJS.get($sortable_element[0]);
     assert(sortable_instance !== undefined);
@@ -449,11 +461,6 @@ function maybe_toggle_alphabetize_button($sortable_element: JQuery): void {
     const alphabetized_array = get_alphabetized_profile_field_choices(sortable_instance);
     const is_alphabetized = _.isEqual(choices_array, alphabetized_array);
 
-    // This is safe because only one profile field modal is open at a time ever.
-    const $alphabetize_choices_button = $(
-        ".profile-field-choices-wrapper > button.alphabetize-choices-button",
-    );
-
     if (!is_alphabetized) {
         $alphabetize_choices_button.prop("disabled", false);
         return;
@@ -586,7 +593,7 @@ function open_edit_form_modal(this: HTMLElement): void {
             "button.delete-choice",
             function (this: HTMLElement) {
                 delete_choice_row_for_edit(this, $profile_field_form, field);
-                maybe_toggle_alphabetize_button($edit_profile_field_choices_container);
+                maybe_toggle_alphabetize_button($edit_profile_field_choices_container, false);
             },
         );
         $profile_field_form.on(
@@ -855,7 +862,7 @@ function set_up_select_field(): void {
     );
     $profile_field_choices.on("click", "button.delete-choice", function (this: HTMLElement) {
         delete_choice_row(this);
-        maybe_toggle_alphabetize_button($profile_field_choices);
+        maybe_toggle_alphabetize_button($profile_field_choices, false);
     });
     $("#profile_field_choices_row").on(
         "click",
```